python/semantic_kernel/connectors/memory/chroma/chroma.py

```
@@ -0,0 +1,372 @@
+            client=client,
+            managed_client=managed_client,
+            **kwargs,
+        )
+
+    def _get_collection(self) -> Collection:
+        try:
+            return self.client.get_collection(name=self.collection_name)
+        except Exception as e:
+            raise RuntimeError(f"Failed to get collection {self.collection_name}") from e
+
+    @override
+    async def does_collection_exist(self, **kwargs: Any) -> bool:
+        """Check if the collection exists."""
+        try:
+            self.client.get_collection(name=self.collection_name)
+            return True
+        except Exception:
+            return False
+
+    @override
+    async def create_collection(self, **kwargs: Any) -> None:
+        """Create the collection.
+
+        Sets the distance function if specified in the data model definition.
+
+        Args:
+            kwargs: Additional arguments are passed to the metadata parameter of the create_collection method.
+
+        """
+        if self.data_model_definition.vector_fields and self.data_model_definition.vector_fields[0].distance_function:
+            if self.data_model_definition.vector_fields[0].distance_function not in DISTANCE_FUNCTION_MAP:
+                raise VectorStoreInitializationException(
+                    f"Distance function {self.data_model_definition.vector_fields[0].distance_function} is not "
+                    "supported."
+                )
+            kwargs["hnsw:space"] = DISTANCE_FUNCTION_MAP[self.data_model_definition.vector_fields[0].distance_function]
+        if kwargs:
+            self.client.create_collection(name=self.collection_name, metadata=kwargs)
+        else:
+            self.client.create_collection(name=self.collection_name)
+
+    @override
+    async def delete_collection(self, **kwargs: Any) -> None:
+        """Delete the collection."""
+        try:
+            self.client.delete_collection(name=self.collection_name)
+        except ValueError:
```

Do we want to swallow the exception?

---

Yeah that's when you delete when there is no collection

---

clarified it now